python/pyspark/sql/session.py

```diff
@@ -394,6 +394,36 @@ def enableHiveSupport(self) -> "SparkSession.Builder":
             """
             return self.config("spark.sql.catalogImplementation", "hive")
 
+        def create(self) -> "SparkSession":
+            """Creates a new SparkSession. Can only be used in the context of Spark Connect
+            and will throw an exception otherwise.
+
+            .. versionadded:: 3.5.0
+
+            Returns
+            -------
+            :class:`SparkSession`
+            """
+            opts = dict(self._options)
+            if "SPARK_REMOTE" in os.environ or "spark.remote" in opts:
+                from pyspark.sql.connect.session import SparkSession as RemoteSparkSession
+
+                url = opts.get("spark.remote", os.environ.get("SPARK_REMOTE"))
```

Comment:
```suggestion
                url = opts.get("spark.remote", os.environ.get("SPARK_REMOTE", None))
                if url is None:
                    raise RuntimeError("Spark Connect URL to the server was not found.")
```

Reply:
This condition would never evaluate to `url is None` because in line 408 we check for the presence of either `spark.remote` or `SPARK_REMOTE`.